messaging/src/main/java/org/axonframework/messaging/QualifiedName.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.StringUtils;
+
+import java.io.Serializable;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.axonframework.common.BuilderUtils.assertNonEmpty;
+import static org.axonframework.common.BuilderUtils.assertThat;
+
+/**
+ * Interface describing a qualified name, providing space for a {@link #namespace() namespace},
+ * {@link #localName() local name}, and {@link #revision() revision}.
+ * <p>
+ * Useful to provide clear names to {@link Message Messages}, {@link MessageHandler MessageHandlers}, and other
+ * components that require naming.
+ * <p>
+ * Note that all characters <b>except</b> for the semicolon ({@code :}) are allowed for any of the three parameters. The
+ * semicolon acts as a delimiter for the {@link #toSimpleString()} and thus is a special character delimited on in the
```

Comment:
Usually (like in SQL) we talk about reserved characters

```suggestion
 * semicolon acts as a delimiter for the {@link #toSimpleString()} and thus is a reserved character.
```